src/main/java/org/mskcc/cmo/metadb/service/impl/ValidRequestCheckerImpl.java

```diff
@@ -0,0 +1,286 @@
+        }
+        if (sampleMap.get("specimenType") != "CellLine"
+                || sampleMap.get("specimenType") != "PDX"
+                || sampleMap.get("specimenType") != "Xenograft"
+                || sampleMap.get("specimenType") != "XenograftDerivedCellLine"
+                || sampleMap.get("specimenType") != "Organoid") {
+            return hasCmoSampleClass(sampleMap);
+        }
+        if (sampleMap.get("specimenType") == "exosome"
+                || sampleMap.get("specimenType") == "cfDNA") {
+            return hasSampleOrigin(sampleMap);
+        }
+        return true;
+    }
+
+    private boolean hasCmoSampleClass(Map<String, String> sampleMap) {
+        if (Strings.isBlank(sampleMap.get("cmoSampleClass"))) {
+            return false;
+        }
+        if (!EnumUtils.isValidEnumIgnoreCase(CmoSampleClass.class, sampleMap.get("cmoSampleClass"))) {
+            return false;
+        }
+        return true;
+    }
+
+    private boolean hasSampleOrigin(Map<String, String> sampleMap) {
+        if (Strings.isBlank(sampleMap.get("sampleOrigin"))) {
+            return false;
+        }
+        if (EnumUtils.isValidEnumIgnoreCase(SampleOrigin.class, sampleMap.get("sampleOrigin"))) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * hasSampleType ensures all variables required to retrieve
+     * Nucleid Acid Abbreviation are available
+     * 
+     * <p><ul>
+     * <li>if sampleType is null empty, checks NAtoExtract
+     * <li>if sampleType is "Pooled Library", checks recipe
+     * <li>Else returns true if sampleType has a valid value
+     * </ul>
+     * @param sampleMap
+     * @return
+     */
+    private boolean hasSampleType(Map<String, String> sampleMap) {
+        if (Strings.isBlank(sampleMap.get("sampleType"))) {
+            return hasNAtoExtract(sampleMap);
+        }
+        if (sampleMap.get("sampleType") == "Pooled Library") {
+            return hasRecipe(sampleMap);
+        }
+        if (EnumUtils.isValidEnumIgnoreCase(SampleType.class, sampleMap.get("sampleType"))) {
+            return true;
+        }
+        return false;
+    }
+
+    private boolean hasRecipe(Map<String, String> sampleMap) {
+        if (Strings.isBlank(sampleMap.get("recipe"))) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * missing this field from sampleManifest, returns true for testing
+     * @param sampleMap
+     * @return
+     */
+    private boolean hasNAtoExtract(Map<String, String> sampleMap) {
+        if (Strings.isBlank(sampleMap.get("NAtoExtract"))) {
+            return true;
+        }
+        return true;
+    }
+
+    /**
+     * missing this field from sampleManifest, returns true for testing
+     * @param sampleMap
+     * @return
+     */
+    private boolean hasNormalizedPatientId(Map<String, String> sampleMap) {
+        if (Strings.isBlank(sampleMap.get("normalizedPatientId"))) {
```

Comment:
same comment here except the field name already matches the expected field name, just need to extract it from "cmoSampleFields" within the sample map

Reply:
also same comment here about checking the sampleMap for "cmoSampleIdFields", etc.